### PR TITLE
fix(common): warn if using ngSrcset without a configured image loader

### PR DESCRIPTION
**Pull request by IvanNavarroCode**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #48655 

## What is the new behavior?

A warn is logged when the attribute `ngSrcset` is used but no loader is configured.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by Ivnosing** [comment]
Is there anything else I could do for pipelines that threw an error or failed? 🙂 I'm not sure what are the issues there.

**Review by hexiaokang — APPROVED** [review]

**Comment by alan-agius4** [comment]
Looks like one of the tests is failing
```
//packages/core/test/bundling/image-directive:protractor_tests_chromium  FAILED in 2 out of 2 in 13.0s

Failures:
1) NgOptimizedImage directive should not warn if there is no oversized image
  Message:
    Expected 1 to equal 0.
  Stack:
    Error: Failed expectation
        at /b/f/w/bazel-out/k8-fastbuild/bin/packages/core/test/bundling/packages/core/test/bundling/image-directive/e2e/oversized-image/oversized-image.e2e-spec.ts:19:25
        at Generator.next (<anonymous>)
        at fulfilled (/b/f/w/bazel-out/k8-fastbuild/bin/packages/core/test/bundling/image-directive/protractor_tests_chromium.sh.runfiles/angular/packages/core/test/bundling/image-directive/protractor_tests_bundle_spec.js:35:26)
        at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

File: packages/common/src/directives/ng_optimized_image/ng_optimized_image.ts [hunk 1]
```diff
@@ -953,3 +954,16 @@ function assertNotMissingBuiltInLoader(ngSrc: string, imageLoader: ImageLoader)
...
+
+/**
+ * Warns if ngSrcset is present and no loader is configured (i.e. the default one is being used).
+ */
+function assertNoNgSrcsetWithoutLoader(dir: NgOptimizedImage, imageLoader: ImageLoader) {
+  if (dir.ngSrcset && imageLoader === noopImageLoader) {
+    console.warn(formatRuntimeError(
+        RuntimeErrorCode.NGSRCSET_WITHOUT_LOADER,
+        `${imgDirectiveDetails(dir.ngSrc)} the \`ngSrcset\` attribute is present but ` +
+            `no image loader is configured (i.e. the default one is being used).` +
```
Comment (AndrewKushnir) [inline packages/common/src/directives/ng_optimized_image/ng_optimized_image.ts:966]:
```suggestion
            `no image loader is configured (i.e. the default one is being used), ` +
            `which would result in the same image being used for all configured sizes. ` +
```

**Review by AndrewKushnir — COMMENTED** [review]
@Ivnosing thanks for the PR 👍  The changes look good, I've just left a minor comment.

**Comment by AndrewKushnir** [comment]
@Ivnosing as @alan-agius4 mentioned above, there is a CI failure, potentially related to these changes, where there is an extra console warning in one of the tests. I believe you may need to update the  `packages/core/test/bundling/image-directive/e2e/oversized-image/oversized-image.ts` file and configure a loader there, for example:

```
providers: [
  {provide: IMAGE_LOADER, useFactory: () => (config: ImageLoaderConfig) => config.src}
]
```

Please let us know if you have any questions.

**Comment by AndrewKushnir** [comment]
@Ivnosing thanks for addressing the feedback! When you get a chance, could you please merge all commits into one, so that we can proceed with the final stages of the code review process? Thank you.

**Comment by Ivnosing** [comment]
@AndrewKushnir @alan-agius4 thanks for the feedback and support!

**Review by AndrewKushnir — APPROVED** [review]

**Review by AndrewKushnir — APPROVED** [review]
Reviewed-for: public-api

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:504594147:BASE:504594202:1674670308431:ef2adbd).

**Review by jessicajaniuk — COMMENTED** [review]
reviewed-for: public-api

**Review by jessicajaniuk — APPROVED** [review]
reviewed-for: public-api

**Comment by AndrewKushnir** [comment]
Caretaker note: unrelated presubmit failures, the PR is ready for merge.

**Comment by jessicajaniuk** [comment]
This PR was merged into the repository by commit a055196c55c57de281ff7167cd4dfb3df783c337.

**Pull request closed without merging**